### ISSUE TITLE
[Bug] Chapter transitions not skipped for single-page series

### ISSUE THREAD

**Issue by Kabutak** [tachiyomiorg/tachiyomi#3079]
### Device information
* Tachiyomi version: Stable 0.9.2 (existed on Stable 0.9.0 as well)
* Android version: 7.0
* Device: Samsung Galaxy S5 Neo

## Steps to reproduce
1. Start reading a downloaded series which has only one page per chapter (e.g. Cyanide and Happiness or another webcomic)
2. Read through several chapters (with Always show chapter transitions off)

### Expected behavior
Chapter transitions should be skipped, since each page is downloaded and loaded before changing to the next one.

### Actual behavior
The transition is skipped the first time, but after that will not be skipped unless the user returns to the chapter list and hits resume again, upon which it will again skip only one transition. It also skips if you flip between two chapters after the initial transition, but that is intended behaviour and unrelated.

### Other details
Related to #2728, which was resolved previously, but this is a different issue.

It is likely a race condition or similar related to the fact that the first page is also the last page, so it may be skipping preloading or another trigger related to the transitions, but only when already reading. When resuming it seems to properly detect that it is the last page and it should begin to load the next chapter. The check is probably in the actual chapter transition itself.

**Comment by arkon** [comment]
I dug a little bit into it. It seems like the next (or previous) chapters aren't yet in the state of `ReaderChapter.State.Loaded` since they're not preloaded.

Haven't actually figured out how to handle that though.

**Comment by scb261** [comment]
I hope that I understood it correctly (wanted to try to write a fix, but I don't fully understand the code). Took several hours to download and launch all this stuff. I will write about https://github.com/inorichi/tachiyomi/blob/dev/app/src/main/java/eu/kanade/tachiyomi/ui/reader/viewer/pager/PagerViewer.kt , but I guess it also applies to WebtoonViewer.kt

When first (and at the same time the only one) page is loaded, at the line 163 expression `page.chapter == adapter.currentChapter` is `false`. Not sure what it does (checks if it's still the same chapter?), but it prevents preload.

When I removed this part (so it would be only `inPreloadRange && allowPreload`) and looked at the variables, I understood that the one that we want to preload is next to the next chapter (since the one that will appear on screen right now is considered next for now). And so removing `page.chapter == adapter.currentChapter` doesn't help because it just tries to preload next one twice.

I think, at the line 168 you need to add next code:
```
else if ([next chapter page (similar to line 147)] && [nextChapter.numberOfPages == 1]) {
    [preload nextChapter.nextChapter]
}
```

Also I didn't see animation of loading while I was debugging, but in stable 0.9.2 it's present ([screenshot](https://imgur.com/ApLr0ZR)). Is it normal? Behavior of preloading seems to be the same.

**Comment by BartTerpstra** [comment]
as someone who likes to read KillSixBillionDemons and "Stand Still, Stay Silent", this bothers me to no end.
i can't take it anymore.
i'm giving it a shot.

**Comment by Saud-97** [comment]
Closed by #6998?
